### DISCUSSION TITLE
jwt + mongoose = timeout

### DISCUSSION THREAD

**Discussion by TheDummi** [vercel/next.js#58327] [Help]
### Summary

I'm making an auth login, and on my dev side it all works perfectly fine, but as soon as it's pushed to production upon login in throws a timeout on mongoose. 

I'm not too familiar with both jwt and nextauth, I've been on this issue for quite a while now, I've tried setting a NEXTAUTH_URL in the env, adding an extra connection to mongo as I saw a lot of issues regarding this had to do with not having a connection to mongo, added extra options to the mongoose.connect(). The mongo IP is set to 0.0.0.0 so IP validation shouldn't be a problem either.

 

### Additional information

```js
This is the error


[next-auth][error][JWT_SESSION_ERROR] 
https://next-auth.js.org/errors#jwt_session_error Operation `users.findOne()` buffering timed out after 10000ms {
  message: 'Operation `users.findOne()` buffering timed out after 10000ms',
  stack: 'MongooseError: Operation `users.findOne()` buffering timed out after 10000ms\n' +
    '    at Timeout.<anonymous> (/var/task/node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js:185:23)\n' +
    '    at listOnTimeout (node:internal/timers:569:17)\n' +
    '    at process.processTimers (node:internal/timers:512:7)',
  name: 'MongooseError'
```

**Comment by ashrth** [comment]
Might be that the page is rendering first and data is being fetched late.
